fvp.mk

```diff
@@ -38,22 +38,22 @@ all-clean: arm-tf-clean busybox-clean edk2-clean optee-os-clean \
 ################################################################################
 # ARM Trusted Firmware
 ################################################################################
-arm-tf: optee-os edk2
+ARM_TF_FLAGS ?= \
 	CFLAGS="-O0 -gdwarf-2" \
-	CROSS_COMPILE="$(CCACHE)$(AARCH64_NONE_CROSS_COMPILE)" \
 	BL32=$(OPTEE_OS_BIN) \
 	BL33=$(EDK2_BIN) \
-	make -C $(ARM_TF_PATH) \
-	       -j`getconf _NPROCESSORS_ONLN` \
-	       DEBUG=0 \
-	       FVP_TSP_RAM_LOCATION=tdram \
-	       FVP_SHARED_DATA_LOCATION=tdram \
-	       PLAT=fvp \
-	       SPD=opteed \
-	       all fip
+	DEBUG=0 \
+	FVP_TSP_RAM_LOCATION=tdram \
+	FVP_SHARED_DATA_LOCATION=tdram \
+	PLAT=fvp \
+	SPD=opteed
+
+arm-tf: optee-os edk2
+	CROSS_COMPILE="$(CCACHE)$(AARCH64_NONE_CROSS_COMPILE)" \
+	make -C $(ARM_TF_PATH) $(ARM_TF_FLAGS) all fip
```

Comment:
This is not correct, because for ARM TF some flags such as `CFLAGS` have to be passed as environment variables (otherwise they override the values set in the Makefile and you lose some required flags), while some others can indeed be passed as make arguments.

With the above I get the following error:
```
  DEPS    build/hikey/release/bl31/plat_io_storage.d
plat/hikey/plat_io_storage.c:32:26: fatal error: arch_helpers.h: No such file or directory
 #include <arch_helpers.h>
```

Here is what works for me:
```
ARM_TF_FLAGS ?= \
	CROSS_COMPILE="$(CCACHE)$(AARCH64_CROSS_COMPILE)" \
	BL32=$(OPTEE_OS_BIN) \
	BL33=$(EDK2_BIN) \
	NEED_BL30=yes \
	BL30=$(MCUIMAGE_BIN) \
	DEBUG=$(DEBUG) \
	PLAT=hikey \
	SPD=opteed

ARM_TF_EXPORTS ?= CFLAGS="-O0 -gdwarf-2"

arm-tf: mcuimage optee-os edk2
	export $(ARM_TF_EXPORTS) ; \
	$(MAKE) -C $(ARM_TF_PATH) $(ARM_TF_FLAGS) all fip

arm-tf-clean:
	export $(ARM_TF_EXPORTS) ; \
	$(MAKE) -C $(ARM_TF_PATH) $(ARM_TF_FLAGS) clean
```